src/main/kotlin/no/nav/familie/ef/sak/service/TilkjentYtelseService.kt

```diff
@@ -43,11 +39,21 @@ class TilkjentYtelseService(private val oppdragClient: OppdragClient,
         return tilkjentYtelse.tilDto()
     }
 
-    @Transactional
+    fun opprettTilkjentYtelse(behandling: Behandling): TilkjentYtelse {
+        val nyTilkjentYtelse = hentTilkjentYtelseFraBehandlingId(behandlingId = behandling.id)
+        return opprettTilkjentYtelse(nyTilkjentYtelse, behandling)
+    }
+
+
     fun opprettTilkjentYtelse(tilkjentYtelseDTO: TilkjentYtelseDTO): TilkjentYtelse {
         val nyTilkjentYtelse = tilkjentYtelseDTO.tilTilkjentYtelse()
-
         val behandling = behandlingService.hentBehandling(nyTilkjentYtelse.behandlingId)
+        return opprettTilkjentYtelse(nyTilkjentYtelse, behandling)
+    }
+
+    @Transactional
```

Comment:
Nå husker jeg ikke helt dette med `@Transactional`, men hvordan funker det når det blir kalt fra en i "samme klasse"? Må den som kaller denne også ha `@Transactional` ?

Er vel Spring-proxy-magien som gjør at eksterne klasser kaller via transactional-proxy og inn i denne, men internt bruker den ikke proxyen?

Reply:
"Any self-invocation calls will not start any transaction, even if the method has the @Transactional annotation"  så hvis man har @transactional metode1 som kaller på @transactional metode2 i samme klass sååå kommer det inte leda til en transaction ?

Reply:
Så længe metod1 ær kallad på utifrån så blir det en transaction.
Men en metod A som kallar på transaction metod B  i samma klass så blir det inte en transaction